src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -204,11 +206,31 @@ private void CacheInstanceDiscoveryMetadata(InstanceDiscoveryMetadataEntry metad
 
         private async Task<Uri> BuildAuthorityWithRegionAsync(Uri canonicalAuthority, RequestContext requestContext)
         {
+            string regionToUse = requestContext.ServiceBundle.Config.AuthorityInfo.RegionToUse;
+
             if (s_region.IsNullOrEmpty())
             {
-                s_region = await GetRegionAsync(requestContext).ConfigureAwait(false);
+                try
+                {
+                    s_region = await GetRegionAsync(requestContext).ConfigureAwait(false);
+
+                    if (!regionToUse.IsNullOrEmpty())
+                    {
+                        requestContext.ApiEvent.IsValidUserProvidedRegion = s_region.Equals(regionToUse);
+                    }
+                }
+                catch (MsalServiceException e)
+                {
+                    if (regionToUse.IsNullOrEmpty())
+                    {
+                        throw e;
```

Comment:
>e [](start = 30, length = 1)

remove

Reply:
This ensures that we only throw an exception when we fail to auto detect and we don't have a user provided region.